lib/matplotlib/sphinxext/plot_directive.py

```diff
@@ -372,9 +378,8 @@ def split_code_at_show(text):
         `{{ fmt }} <{{ dest_dir }}/{{ img.basename }}.{{ fmt }}>`__
         {%- endfor -%}
         )
-      {%- endif -%}
-
-      {{ caption }}
+      {%- endif %}
+{{ caption }}  {# appropriate leading whitespace added beforehand #}
```

Comment:
Stripping trailing whitespace here causes the extra whitespace from before the `{% if` it matches to be effectively prepended to the caption, causing a bug.

Stripping that earlier whitespace using `{% - if` doesn't work because then Jinja strips the newline separating the options from the content. 

So I settled for this ugly solution which puts "{{ caption }}" up against the side. 

The other option would be to remove the artificially-inserted leading whitespace from the first line of the "caption" (L758), but this would technically be changing API for users who already have made custom templates for use the with plot_directive.